cylc/flow/task_pool.py

```diff
@@ -597,17 +599,15 @@ def release_runahead_task(
             # implicit prev-instance parent
             return
 
-        if not itask.flow_nums:
-            # No reflow
-            return
-
-        if not runahead_limit_point:
-            return
-
-        # Autospawn successor of itask if parentless.
-        n_task = self.spawn_successor(itask)
-        if n_task and n_task.point <= runahead_limit_point:
-            self.release_runahead_task(n_task, runahead_limit_point)
+        # Autospawn successor of itask if parentless and reflowing.
+        if itask.flow_nums:
+            n_task = self.spawn_successor_if_parentless(itask)
+            if (
+                n_task and
+                runahead_limit_point and
+                n_task.point <= runahead_limit_point
+            ):
+                self.release_runahead_task(n_task, runahead_limit_point)
```

Comment:
I think this is identical to #4621?

Reply:
No, according to my tests 4621 did not actually fix the problem… but this does, so it must be different!

Reply:
The difference in the logic is that here it will still `spawn_successor_if_parentless()` regardless of whether `runahead_limit_point` is `None` or not

Reply:
Ah gotcha, good, helps me understand.